docs/src/pages/components/data-grid/filtering/CustomRatingOperator.js

```diff
@@ -73,17 +73,23 @@ const RatingOnlyOperators = [
 
 export default function CustomRatingOperator() {
   const { data } = useDemoData({ dataSet: 'Employee', rowLength: 100 });
+  const columns = [...data.columns];
```

Comment:
Shouldn't this just work, without having to copy the columns?

Reply:
So demos are not sandboxed properly and they share the instance of the app when they are rendered in a page.

Reply:
> Shouldn't this just work

@DanailH No, it shouldn't work because renders needs to be **immutable**. It's an important constraint that strict mode and concurrence mode sets.

Reply:
> So demos are not sandboxed properly

@dtassone I have open #979 to go a step deeper in the resolution of the root issue. Real apps aren't sandboxed, I believe it's thanks to the documentation not being sandboxed that we could have spotted the issue, making the demos sandboxed would be a step backward.